test/unit/shared/access_permissions_test_base.rb

```diff
@@ -21,10 +21,15 @@ def check_routes(app_routes, skipped_actions)
           controller = "#{r.defaults[:controller]}_controller".classify.constantize
           filters    = controller.send(:_process_action_callbacks)
 
-          # Pass if the controller deliberately only permit admins (e.g. SettingsController)
-          if filters.select { |f| f.filter == :require_admin }.empty?
-            assert_not_equal [], Foreman::AccessControl.permissions.select { |p| p.actions.include? path }
-          end
+          permission_required = true
+
+          # Pass if the controller only permit admins (e.g. SettingsController)
+          permission_required = false if filters.select { |f| f.filter == :require_admin }.any?
+
+          # Pass if the controller deliberately skips login requirement
+          permission_required = false if controller < ApplicationController && filters.select { |f| f.filter == :require_login }.empty?
```

Comment:
not sure why here we check for `ApplicationController` but not for the other one?
Also we could just check both in one run using `f.filter.in? [:require_login, :require_admin]`

Reply:
The API controllers do not have a `require_login` action as they don't inherit from `ApplicationController`. That's why I had to split that up. The second check is only valid if the first condition is true.

Reply:
oh, I see, makes sense.